src/cow-react/common/hooks/useTokenBySymbolOrAddress.ts

```diff
@@ -0,0 +1,20 @@
+import { Token } from '@uniswap/sdk-core'
+import { useMemo } from 'react'
+import { useAllTokensList } from '@src/cow-react/common/hooks/useAllTokensList'
+
+export function useTokenBySymbolOrAddress(symbolOrAddress?: string | null): Token | null {
+  const tokens = useAllTokensList()
+
+  return useMemo(() => {
+    if (!symbolOrAddress) {
+      return null
+    }
+
+    return (
+      tokens.find(
+        (item) => item.address.toLowerCase() === symbolOrAddress.toLowerCase() || item.symbol === symbolOrAddress
+      ) || null
+    )
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Why do you need this?
Maybe I'm not seeing something

Reply:
Thanks! It's redundant